### PR TITLE
[Poetry] Hide poem dropdown in view mode

### PR DESCRIPTION
**Pull request by ajpal**
We don't want to allow other users to change the selected poem in the dropdown when they're viewing someone else's project
Also hide the poem dropdown in the costume tab.

Before
image
image

After
image
image

Side note- we might want to also do this for dance party? right now you *can* change the song in view mode.

### CODE REVIEW

File: apps/src/p5lab/P5LabVisualizationHeader.jsx [hunk 1]
```diff
@@ -58,11 +59,13 @@ class P5LabVisualizationHeader extends React.Component {
 
   render() {
     const {interfaceMode, allowAnimationMode} = this.props;
+    const shouldShowPoemSelector =
```
Comment (maddiedierker) [inline apps/src/p5lab/P5LabVisualizationHeader.jsx]: nit: prefer this calculation in a function to simplify the render function

**Review by maddiedierker — COMMENTED** [review]
we have a related ticket that may be scheduled for next sprint -- [STAR-1803](https://codedotorg.atlassian.net/browse/STAR-1803) (only show "select poem" dropdown in code mode). maybe sync with @KylieModen on whether to include that as part of this PR? (i ask because the ticket has a "?" so the HoC v-team may decide we don't want to do this work.)

**Review by maddiedierker — APPROVED** [review]

**Review by katleiahramos — APPROVED** [review]
Pulled & Tested ✅ 

Is this something we'd want a UI / eyes test for?

**Comment by ajpal** [comment]
> Is this something we'd want a UI / eyes test for?

Yeah, ideally. I'm planning to add test coverage after everything the 11/12 deadline

**Pull request merged**